io/zenoh-links/zenoh-link-quic/src/utils.rs

```diff
@@ -271,93 +265,78 @@ impl TlsClientConfig {
         // Allows mixed user-generated CA and webPKI CA
         tracing::debug!("Loading default Web PKI certificates.");
         let mut root_cert_store = RootCertStore {
-            roots: webpki_roots::TLS_SERVER_ROOTS
-                .iter()
-                .map(|ta| ta.to_owned())
-                .map(|ta| {
-                    OwnedTrustAnchor::from_subject_spki_name_constraints(
-                        ta.subject.to_vec(),
-                        ta.subject_public_key_info.to_vec(),
-                        ta.name_constraints.map(|nc| nc.to_vec()),
-                    )
-                })
-                .collect(),
+            roots: webpki_roots::TLS_SERVER_ROOTS.to_vec(),
         };
 
         if let Some(custom_root_cert) = load_trust_anchors(config)? {
             tracing::debug!("Loading user-generated certificates.");
-            root_cert_store.roots.extend(custom_root_cert.roots);
+            root_cert_store.extend(custom_root_cert.roots);
         }
 
+        // Install rustls provider
+        rustls::crypto::ring::default_provider()
+            .install_default()
+            .ok();
```

Comment:
I think it's better to return an error if this operation fails.
```suggestion
            .map_err(|err| zerror!("Error installing TLS crypto provider: {err}."))?;
```